tests/pass-dep/libc/pthread-threadname.rs

```diff
@@ -55,8 +55,15 @@ fn main() {
         // But the system is limited -- make sure we successfully set a truncation.
         let mut buf = vec![0u8; long_name.len() + 1];
         assert_eq!(get_thread_name(&mut buf), 0);
+
+        // A specific case of glibc which requires the buffer.to be at least 15 bytes long
+        // excluding a null terminator;
+        #[cfg(target_os = "linux")]
+        assert_ne!(get_thread_name(&mut buf[..15]), 0);
```

Comment:
This here should return ERANGE also on Solarish -- and it passes already without your change. We already return ERANGE when the thread name is longer than the buffer.

Reply:
Not a commit I wanted to open a PR for. Forgot to push it.

Reply:
Initially I made what you have seen and pushed it. Then checked the diff and improved the test, but forgot to force push it, so made a soft reset on top of the current changes to avoid breaking the history.

There was no test for a shorter name which is suitable to verify the `glibc` behavior.